ert_shared/ensemble_evaluator/prefect_ensemble/prefect_ensemble.py

```diff
@@ -0,0 +1,317 @@
+                        status="Unknown",
+                        name=stage["name"],
+                    )
+                )
+            reals.append(_Realization(iens=iens, stages=stages, active=True))
+        return reals
+
+    @staticmethod
+    def _on_task_failure(task, state, url):
+        with Client(url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_FM_STEP_FAILURE,
+                    "source": f"/ert/ee/{task.get_ee_id()}/real/{task.get_iens()}/stage/{task.get_stage_id()}/step/{task.get_step_id()}",
+                    "datacontenttype": "application/json",
+                },
+                {"stderr": state.message},
+            )
+            c.send(to_json(event).decode())
+
+    def get_id(self, iens, stage_name, step_name=None, job_index=None):
+        real = next(real for real in self._reals if real.get_iens() == iens)
+        stage = next(
+            stage for stage in real.get_stages() if stage.get_name() == stage_name
+        )
+        if step_name is None:
+            return stage.get_id()
+        step = next(step for step in stage.get_steps() if step.get_name() == step_name)
+        if job_index is None:
+            return step.get_id()
+        job = step.get_jobs()[job_index]
+        return job.get_id()
+
+    def get_ordering(self, iens):
+        table_of_elements = []
+        for stage in self.config["stages"]:
+            for step in stage["steps"]:
+                jobs = [
+                    {
+                        "id": self.get_id(
+                            iens,
+                            stage_name=stage["name"],
+                            step_name=step["name"],
+                            job_index=idx,
+                        ),
+                        "name": job.get("name"),
+                        "executable": job.get("executable"),
+                        "args": job.get("args", []),
+                    }
+                    for idx, job in enumerate(step.get("jobs", []))
+                ]
+                table_of_elements.append(
+                    {
+                        "iens": iens,
+                        "stage_name": stage["name"],
+                        "stage_id": self.get_id(iens, stage_name=stage["name"]),
+                        "step_id": self.get_id(
+                            iens, stage_name=stage["name"], step_name=step["name"]
+                        ),
+                        **step,
+                        "jobs": jobs,
+                    }
+                )
+
+        produced = set()
+        ordering = []
+        while table_of_elements:
+            temp_list = produced.copy()
+            for element in table_of_elements:
+                if set(element.get("inputs", [])).issubset(temp_list):
+                    ordering.append(element)
+                    produced = produced.union(set(element["outputs"]))
+                    table_of_elements.remove(element)
```

Comment:
I think this can be done more efficiently, but we can create a separate issue for this.

Reply:
I am sure it can be done in a more efficient way. But I agree it should be done in a separate issue.